src/main/java/appeng/client/gui/implementations/SpatialAnchorScreen.java

```diff
@@ -0,0 +1,257 @@
+
+        final int gx = (this.width - this.xSize) / 2;
+        final int gy = (this.height - this.ySize) / 2;
+
+        this.tooltip = -1;
+
+        int y = 0;
+        int x = 0;
+        for (int z = 0; z <= SCROLL_ROWS * SCROLL_COLUMNS; z++) {
+            final int minX = gx + 9 + x * SCROLL_ITEM_WIDTH;
+            final int minY = gy + 40 + y * SCROLL_ITEM_HEIGHT;
+
+            if (minX + SCROLL_ITEM_PADDING_X < mouseX && minX + SCROLL_ITEM_WIDTH - SCROLL_ITEM_PADDING_X > mouseX) {
+                if (minY + SCROLL_ITEM_PADDING_Y < mouseY
+                        && minY + SCROLL_ITEM_HEIGHT - SCROLL_ITEM_PADDING_Y > mouseY) {
+                    this.tooltip = z;
+                    break;
+                }
+            }
+
+            x++;
+
+            if (x >= SCROLL_COLUMNS) {
+                y++;
+                x = 0;
+            }
+        }
+
+        super.render(matrixStack, mouseX, mouseY, btn);
+    }
+
+    @Override
+    public void drawFG(MatrixStack matrixStack, final int offsetX, final int offsetY, final int mouseX,
+            final int mouseY) {
+
+        if (this.overlayToggle != null) {
+            this.overlayToggle.set(this.container.getOverlayMode());
+        }
+
+        this.font.drawString(matrixStack, this.getGuiDisplayName(GuiText.SpatialAnchor.text()).getString(), 8, 6,
+                4210752);
+
+        String usedPower = GuiText.SpatialAnchorUsedPower
+                .text(Platform.formatPowerLong(this.container.powerConsumption * 100, true)).getString();
+        this.font.drawString(matrixStack, usedPower, 13, 16, 4210752);
+        this.font.drawString(matrixStack,
+                GuiText.SpatialAnchorLoadedChunks.text(this.container.loadedChunks).getString(), 13, 26, 4210752);
+
+        this.font.drawString(matrixStack, this.getGuiDisplayName(GuiText.SpatialAnchorStatistics.text()).getString(), 8,
+                151 - 30, 4210752);
+
+        this.font.drawString(matrixStack,
+                GuiText.SpatialAnchorAllLoaded.text(this.container.allLoadedChunks, this.container.allLoadedWorlds)
+                        .getString(),
+                13, 151 - 20, 4210752);
+
+        this.font.drawString(matrixStack,
+                GuiText.SpatialAnchorAll.text(this.container.allChunks, this.container.allWorlds).getString(), 13,
+                151 - 10, 4210752);
+
+        int x = 0;
+        int y = 0;
+        final int xo = 8;
+        final int yo = 42;
+        final int viewStart = 0;
+        final int viewEnd = viewStart + SCROLL_COLUMNS * SCROLL_ROWS;
+
+        String toolTip = "";
+        int toolPosX = 0;
+        int toolPosY = 0;
+
+        for (int z = viewStart; z < Math.min(viewEnd, this.repo.size()); z++) {
+            final IAEItemStack refStack = this.repo.getReferenceItem(z);
+            if (refStack != null) {
+                ChunkPos chunkPos = new ChunkPos(refStack.getDefinition().getOrCreateTag().getLong("chunk"));
```

Comment:
I don't remember. Did `getDefinition()` always create a new ItemStack?

Reply:
That won't create a new one. But I guess a simple `getTag()` would be better to avoid corrupting the shared registry. The whole thing is just a massive hack to display a list of chunks. With the world overlay now I would even say we should remove it.

Reply:
This code is now completely removed again.